internal/js/modules/k6/browser/common/frame_manager.go

```diff
@@ -682,6 +691,10 @@ func (m *FrameManager) NavigateFrame(frame *Frame, url string, parsedOpts *Frame
 	select {
 	case evt := <-navEvtCh:
 		if e, ok := evt.(*NavigationEvent); ok {
+			if e.err != nil {
+				return nil, e.err
```

Comment:
Here's the key to surface the errors, previously we were ignoring if the `NavigationEvent` was holding an `error`.